src/ray/core_worker/context.cc

```diff
@@ -172,12 +178,32 @@ ObjectIDIndexType WorkerContext::GetNextPutIndex() {
   return GetThreadContext().GetNextPutIndex();
 }
 
+void WorkerContext::MaybeInitializeJobInfo(const JobID &job_id,
+                                           const rpc::JobConfig &job_config) {
+  absl::WriterMutexLock lock(&mutex_);
+  if (current_job_id_.IsNil()) {
+    current_job_id_ = job_id;
+  }
+  if (google::protobuf::util::MessageDifferencer::Equals(job_config_,
```

Comment:
Should we assert that this function is not called with two distinct, non-default job configs?

```
MaybeInitializeJobInfo(..., job_config=A) -> set the job config
... some time later
MaybeInitializeJobInfo(..., job_config=B) -> assert false since this is undefined behavior
```